### DISCUSSION TITLE
Get a nested key as id while validating a nested array

### DISCUSSION THREAD

**Discussion by berrymore** [Respect/Validation#1496] [Q&A] [answered]
```php
<?php

use Respect\Validation\Validator as v;
use Respect\Validation\Exceptions\ValidationException;

$data = [
    'user' => [
       'name' => 'this is a too long name for a user',
    ]
];

try {
    v::key('user', v::key('name', v::stringType()->length(1, 30)))->check($data);
} catch (ValidationException $e) {
    var_dump($e->getId()); // name
}
```

In the exception it returns `name` while I expect `user.name`. Is it possible to achieve this behavior?

**Comment by berrymore** [comment, marked as answer]
```php
// ...

try {
    v::allOf(
        v::keyNested('user.name', v::stringType()->length(1, 30))
    )->check($data);
} catch (ValidationException $e) {
    var_dump($e->getId()); // user.name
}
```

As a workaround, I use this solution

**Reply by henriquemoody** [reply]
Hi @berrymore! Nice that you found a solution for your problem. Could you tell me what's your use case, though?

Asking because `keyNested()` will be deprecated in version 3.0.

**Reply by berrymore** [reply]
Hi @henriquemoody! In my projects, when a validation error happens during the request execution, I usually return a model that contains an array of violations, where a violation has `path` and `message` fields. Since there are nested structures in a request, I'd like to respond with nested paths in violations. For example:

Request
```json
{
  "address": {
    "countryCode": "123"
  }
}
```

Response
```json
{
  "violations": [
    {"path": "address.countryCode", "message": "You should specify a valid ISO-3166 country code"}
  ]
}
```

This way it becomes easy to handle the response on the frontend side to address the error message to the specific input control.

**Reply by henriquemoody** [reply]
Thanks for your answer!

**Reply by henriquemoody** [reply]
Thanks to your comments, I thought better about the error messages and came to the conclusion they were not so helpful when dealing with nested structure. I've been working on some changes, there are a couple of challenges to make it work, I thought I would let you know what I'm doing: https://github.com/Respect/Validation/pull/1507#discussion_r1894786636

I will take a break in the next weeks, I will get back next year, and will try to get the version 3.0 out at the beginning of the year!

**Reply by henriquemoody** [reply]
If you check the pull request I sent you, you will be happy to know that it got merged. It will be together with version 3.0, which I hope to release at the beginning of next year.
